### DISCUSSION TITLE
Python 3.11 support

### DISCUSSION THREAD

**Discussion by Lservi** [facebookresearch/esm#682] [General]
We are using this amazing tool on our server with Python 3.9. Since Python 3.11 significantly improves performance (around a 30% reduction in execution time for many tasks), I was wondering if there are plans to make this tool compatible with Python 3.11.

Thanks for all the work!

**Comment by obragaa** [comment]
Which tool?

**Reply by Lservi** [reply]
Thanks for the fast answer. I was referring to this: "To use the ESMFold model, make sure you start from an environment with python <= 3.9 ."

**Reply by obragaa** [reply]
Hi Lservi,

It seems that this is an important issue that hasn't yet been openly discussed by the developers of the **ESMFold** model. A good way to get a direct response, and perhaps even draw attention to the need for an update, would be to **open an "issue" in the ESMFold GitHub repository**.

In the issue, you can explain how upgrading to **Python 3.11** could benefit performance and ask about their future plans for supporting this version. This not only might give you a more concrete answer but also help other users who may have the same question.

_If you need help crafting your issue, I'm here to assist!_

**Comment by ebetica** [comment]
Unfortunately, no plans to support this :( I believe it was due to one of the dependencies requiring it. If you find that this isn't the case please submit a PR! Though I don't think this repo is maintained anymore...
